src/main/java/nl/thehyve/ocdu/controllers/UploadSessionController.java

```diff
@@ -0,0 +1,54 @@
+package nl.thehyve.ocdu.controllers;
+
+import nl.thehyve.ocdu.models.UploadSession;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.web.bind.annotation.*;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.*;
+
+/**
+ * Created by bo on 4/26/16.
+ */
+@RestController
+public class UploadSessionController {
+
+    @RequestMapping(value="/username", method= RequestMethod.GET)
+    public String username() {
```

Comment:
I am not sure if this is necessary - did you try retrieving user via
 
    ${session.foo}

session object should be accessible inside Thymeleaf template. Maybe you could just get username from there? It would save a Request roundtrip and quite some code if we can retrieve username from there.